### PR TITLE
RFC: temp path parent directory

### PR DESCRIPTION
**Pull request by nolta**
43c7cae adds an option to `mktemp` and `mktempdir` to control which directory the temporary file/directory is created in.

2414e9e is a workaround for #11191.

### CODE REVIEW

**Comment by tkelman** [comment]
These versions are in a `@unix_only` block, so the change from 43c7cae77316e301ebea1280f21109c9001e3a1c will also need to be added to the Windows versions.

**Comment by nolta** [comment]
Thanks, fixed (maybe).

**Comment by tkelman** [comment]
there should probably be at least one test for this in `test/file.jl`

**Comment by nolta** [comment]
Ok, added tests.

**Comment by tkelman** [comment]
Looks good, thanks

File: base/pkg/dir.jl [hunk 1]
```diff
@@ -37,15 +37,16 @@ function init(meta::AbstractString=DEFAULT_META, branch::AbstractString=META_BRA
...
     metadata_dir = joinpath(dir, "METADATA")
     if isdir(metadata_dir)
         info("Package directory $dir is already initialized.")
         Git.set_remote_url(meta, dir=metadata_dir)
         return
     end
+    local temp_dir
     try
-        mkpath(temp_dir)
+        mkpath(dir)
```
Comment (garrison) [inline base/pkg/dir.jl]: Won't this line fail if `~/.julia/v0.4` exists but not `~/.julia/v0.4/METADATA`?

Comment (garrison) [inline base/pkg/dir.jl]: Nevermind, my bad `mkpath != mkdir`

**Pull request merged**